Common/TableProducer/trackselection.cxx

```diff
@@ -54,30 +58,35 @@ struct TrackSelectionTask {
       case 1:
         // Run 3 kAny on 3 IB layers of ITS
         if (isRun3) {
-          globalTracks = getGlobalTrackSelectionITSMatch(TrackSelection::GlobalTrackRun3ITSMatching::Run3ITSibAny);
+          globalTracks = getGlobalTrackSelectionRun3ITSMatch(TrackSelection::GlobalTrackRun3ITSMatching::Run3ITSibAny);
         }
         break;
       case 2:
         // Run 3 kAny on all 7 layers of ITS
         if (isRun3) {
-          globalTracks = getGlobalTrackSelectionITSMatch(TrackSelection::GlobalTrackRun3ITSMatching::Run3ITSallAny);
+          globalTracks = getGlobalTrackSelectionRun3ITSMatch(TrackSelection::GlobalTrackRun3ITSMatching::Run3ITSallAny);
         }
         break;
       case 3:
         // Run 3 kAll on all 7 layers of ITS
         if (isRun3) {
-          globalTracks = getGlobalTrackSelectionITSMatch(TrackSelection::GlobalTrackRun3ITSMatching::Run3ITSall7Layers);
+          globalTracks = getGlobalTrackSelectionRun3ITSMatch(TrackSelection::GlobalTrackRun3ITSMatching::Run3ITSall7Layers);
         }
         break;
-
       default:
         LOG(fatal) << "TrackSelectionTask with undefined cuts. Fix it!";
         break;
     }
-    globalTracksSDD = getGlobalTrackSelectionSDD();
+    globalTracks.SetPtRange(ptMin, ptMax);
+    globalTracks.SetEtaRange(etaMin, etaMax);
 
+    // Extra requirement on the ITS -> Run 2: asking for 1 hit SDD and no hit in SPD
+    globalTracksSDD = getGlobalTrackSelectionSDD();
+    globalTracksSDD.SetPtRange(ptMin, ptMax);
+    globalTracksSDD.SetEtaRange(etaMin, etaMax);
     if (isRun3) {
-      globalTracks.SetTrackType(o2::aod::track::TrackTypeEnum::Track);
+      globalTracks.SetTrackType(o2::aod::track::TrackTypeEnum::Track);    // Requiring that this is a Run 3 track
+      globalTracksSDD.SetTrackType(o2::aod::track::TrackTypeEnum::Track); // Requiring that this is a Run 3 track
```

Comment:
Is the correct? The SDD is for Run 2, isn't it?

Reply:
That's true, but then it means that we have a column that is shipped always false

Reply:
You mean for Run3 ? Yes, I think this is how it is. We shouldn't even need to evaluate that one in Run 3 mode...

Reply:
Ok, perfect!

Reply:
I fixed the requirement and skipped the computation of the flag for Run3

Reply:
Looks good.